xtb/molecule_reader.f90

```diff
@@ -735,7 +730,13 @@ subroutine read_molecule_pdb(mol, unit, status, iomsg)
       iomsg = "could not read in coordinates, last line was: '"//line//"'"
       return
    endif
-
+   j=0
+   do i = minval(pdb(:iatom)%residue_number),maxval(pdb(:iatom)%residue_number)
+      if (any(i .eq. pdb(:iatom)%residue_number)) then
+         j=j+1
+         where (i .eq. pdb(:iatom)%residue_number) list = j
```

Comment:
The logic here would be clearer if we use counter names like `iresidue`/`ifragment`.
Also an `associate` statement could make this more readable:
```fortran
associate(residue_number => pdb(:iatom)%residue_number)
...
end associate
```

Reply:
I tried the associate function but it did not work.